scheduler/src/cook/kubernetes/api.clj

```diff
@@ -391,16 +400,40 @@
                                                      false)
              :else true))))
 
+(defn deep-merge-with
+  "Like merge-with, but merges maps recursively, applying the given fn
+  only when there's a non-map at a particular level.
+  (deepmerge + {:a {:b {:c 1 :d {:x 1 :y 2}} :e 3} :f 4}
+               {:a {:b {:c 2 :d {:z 9} :z 3} :e 100}})
+  -> {:a {:b {:z 3, :c 3, :d {:z 9, :x 1, :y 2}}, :e 103}, :f 4}"
+  [f & maps]
+  (apply
+    (fn m [& maps]
+      (if (every? map? maps)
+        (apply merge-with m maps)
+        (apply f maps)))
+    maps))
+
+(defn add-gpu-model-to-resource-map
+  "Given a map from node-name->resource-type->capacity, perform the following operation:
+  - if the amount of gpus on the node is positive, set the gpus capacity to model->count
+  - if the amount of gpus on the node is 0, set the gpus capacity to an empty map"
+  [gpu-model gpus resource-map]
```

Comment:
Instead of passing a separate argument `gpus`, could this just be:

```suggestion
  [gpu-model {:keys [gpus] :as resource-map}]
```

Reply:
modified and modified `get-consumption` and `get-capacity` to simplify as well